src/app.js

```diff
@@ -260,32 +257,60 @@ class SAFEApp extends EventEmitter {
         let fileSize;
         let lengthToRead = consts.pubConsts.NFS_FILE_END;
         let endByte;
+        let data;
+        let multipart;
+        let rangeIsArray;
 
-        // TODO: how do we handle multipart Reqs
-        if (options && options.range && typeof options.range.start !== 'undefined') {
-          range = options.range;
-          start = range.start;
+        if (options && options.range) {
+          rangeIsArray = Array.isArray(options.range);
           fileSize = await openedFile.size();
-          end = range.end || fileSize - 1;
-
+          range = options.range;
+          multipart = range.length > 1;
+          start = options.range.start || consts.pubConsts.NFS_FILE_START;
+          end = options.range.end || fileSize - 1;
           lengthToRead = (end - start) + 1; // account for 0 index
         }
-        const data = await openedFile.read(start, lengthToRead);
-        const mimeType = mime.getType(nodePath.extname(filePath)) || 'application/octet-stream';
+
+        if (options && options.range && rangeIsArray) {
+          // block handles multipart range requests
+          data = await Promise.all(range.map(async (part) => {
+            const partStart = part.start;
+            const partEnd = part.end || fileSize - 1;
+            const partLengthToRead = (partEnd - partStart) + 1; // account for 0 index
+            const byteSegment = await openedFile.read(partStart, partLengthToRead);
+            const partEndByte = (partEnd === fileSize - 1) ? fileSize - 1 : partEnd;
+            const mimeType = mime.getType(nodePath.extname(filePath))
```

Comment:
It seems you don't need this within the iteration

Reply:
The mimeType variable? That should be different from the parent scope mimeType.

Reply:
But the value you are setting to `mimeType` is always the same as it doesn't depend on any of the values in each iteration, right?